tests/e2e/e2e_test.go

```diff
@@ -0,0 +1,276 @@
+	})
+
+	// TODO: network runner health should imply custom VM healthiness
+	// or provide a separate API for custom VM healthiness
+	// "start" is async, so wait some time for cluster health
+	time.Sleep(2 * time.Minute)
+
+	ctx, cancel := createDefaultCtx()
+	_, err = cli.Health(ctx)
+	cancel()
+	gomega.Expect(err).Should(gomega.BeNil())
+
+	subnetEVMRPCEps = make([]string, 0)
+	blockchainID, logsDir := "", ""
+	k, err := getVMID("subnetevm")
+	gomega.Expect(err).Should(gomega.BeNil())
+
+	// wait up to 5-minute for custom VM installation
+	ctx, cancel = context.WithTimeout(context.Background(), 5*time.Minute)
+done:
+	for ctx.Err() == nil {
+		select {
+		case <-ctx.Done():
+			break done
+		case <-time.After(5 * time.Second):
+		}
+
+		outf("{{magenta}}checking custom VM status{{/}}\n")
+		cctx, ccancel := createDefaultCtx()
+		resp, err := cli.Status(cctx)
+		ccancel()
+		gomega.Expect(err).Should(gomega.BeNil())
+
+		// all logs are stored under root data dir
+		logsDir = resp.GetClusterInfo().GetRootDataDir()
+
+		if v, ok := resp.ClusterInfo.CustomVms[k.String()]; ok {
+			blockchainID = v.BlockchainId
+			outf("{{blue}}subnet-evm is ready:{{/}} %+v\n", v)
+			break done
+		}
+	}
+	gomega.Expect(ctx.Err()).Should(gomega.BeNil())
+	cancel()
+
+	gomega.Expect(blockchainID).Should(gomega.Not(gomega.BeEmpty()))
+	gomega.Expect(logsDir).Should(gomega.Not(gomega.BeEmpty()))
+
+	cctx, ccancel := createDefaultCtx()
+	uris, err := cli.URIs(cctx)
+	ccancel()
+	gomega.Expect(err).Should(gomega.BeNil())
+	outf("{{blue}}avalanche HTTP RPCs URIs:{{/}} %q\n", uris)
+
+	for _, u := range uris {
+		rpcEP := fmt.Sprintf("%s/ext/bc/%s/rpc", u, blockchainID)
+		subnetEVMRPCEps = append(subnetEVMRPCEps, rpcEP)
+		outf("{{blue}}avalanche subnet-evm RPC:{{/}} %q\n", rpcEP)
+	}
+
+	pid := os.Getpid()
+	outf("{{blue}}{{bold}}writing output %q with PID %d{{/}}\n", outputPath, pid)
+	ci := clusterInfo{
+		URIs:     uris,
+		Endpoint: fmt.Sprintf("/ext/bc/%s", blockchainID),
+		PID:      pid,
+		LogsDir:  logsDir,
+	}
+	gomega.Expect(ci.Save(outputPath)).Should(gomega.BeNil())
+
+	b, err := os.ReadFile(outputPath)
+	gomega.Expect(err).Should(gomega.BeNil())
+	outf("\n{{blue}}$ cat %s:{{/}}\n%s\n", outputPath, string(b))
+})
+
+var _ = ginkgo.AfterSuite(func() {
+	switch mode {
+	case "test":
+		outf("{{red}}shutting down cluster{{/}}\n")
+		ctx, cancel := createDefaultCtx()
+		_, err := cli.Stop(ctx)
+		cancel()
+		gomega.Expect(err).Should(gomega.BeNil())
+
+	case "run":
+		outf("{{red}}skipping shutting down cluster{{/}}\n")
+	}
+
+	outf("{{red}}shutting down client{{/}}\n")
+	gomega.Expect(cli.Close()).Should(gomega.BeNil())
+})
+
+var _ = ginkgo.Describe("[basic]", func() {
+	ginkgo.It("can TODO", func() {
+		if mode != "test" {
+			ginkgo.Skip("mode is not 'test'; skipping...")
+		}
+
+		// TODO: e2e tests specific for subnet-evm
+	})
+})
+
+func createDefaultCtx() (context.Context, context.CancelFunc) {
```

Comment:
I think adding a function like this obscures what is actually happening. Since this is a 1 liner anyways, can we just remove this in favor of creating a context with a 2 minute timeout wherever this is called to make it more clear what the default context is?